python/dask_cudf/dask_cudf/io/parquet.py

```diff
@@ -32,14 +32,25 @@ def read_partition(
             columns += index
 
         strings_to_cats = kwargs.get("strings_to_categorical", False)
-        df = cudf.read_parquet(
-            piece.path,
-            engine="cudf",
-            columns=columns,
-            row_group=piece.row_group,
-            strings_to_categorical=strings_to_cats,
-            **kwargs.get("read", {}),
-        )
+        if cudf.utils.ioutils._is_local_filesystem(fs):
```

Comment:
Is there a specific problem with us using the `else` branch even if the file system is local?  If `fs.open` works for both cases, it might be best to avoid unnecessary code.

Reply:
The reason I went with two codepaths is because going via fs.open creates a python host buffer. This approach was about 25% slower than directly passing the filename down for local files. Tested on a 8gb dataset (in memory) 1.08s vs 1.45s

Reply:
Doesn't `fs.open` read into a Python memory allocation though? That is the ~25% slower that @ayushdg reported above, no?

Reply:
Sorry for missing your performance note - Makes sense to me!